core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceFactoryTest.java

```diff
@@ -199,27 +201,111 @@ void testMultiPathFailingService() {
+                .isEqualTo("This is a description from the annotation");
+    }
+
+    @Test
+    void testParameterDescriptionLoadByParentClass() throws NoSuchMethodException {
+        final Parameter parameter = DescriptionAnnotatedTestClass.class.getMethod("testMethod4", String.class)
+                                                                       .getParameters()[0];
+        final DescriptionInfo descriptionInfo = findDescription(parameter);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a super description from the annotation");
+    }
+
+    @Test
+    void testParameterDescriptionLoadByChildClass() throws NoSuchMethodException {
+        final Parameter parameter = DescriptionAnnotatedTestClass.class.getMethod("testMethod5", String.class)
+                                                                       .getParameters()[0];
+        final DescriptionInfo descriptionInfo = findDescription(parameter);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a description from the annotation");
+    }
+
+    @Test
+    void testMethodDescriptionLoad() throws NoSuchMethodException {
+        final Method method = DescriptionAnnotatedTestClass.class.getMethod("testMethod3", String.class);
+        final DescriptionInfo descriptionInfo = findDescription(method);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a description from the annotation");
+    }
+
+    @Test
+    void testMethodDescriptionLoadByParentClass() throws NoSuchMethodException {
+        final Method method = DescriptionAnnotatedTestClass.class.getMethod("testMethod4", String.class);
+        final DescriptionInfo descriptionInfo = findDescription(method);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a super description from the annotation");
+    }
+
+    @Test
+    void testMethodDescriptionLoadByChildClass() throws NoSuchMethodException {
+        final Method method = DescriptionAnnotatedTestClass.class.getMethod("testMethod5", String.class);
+        final DescriptionInfo descriptionInfo = findDescription(method);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a description from the annotation");
+    }
+
+    @Test
+    void testClassDescriptionLoadByParentClass() {
+        final Class<DescriptionAnnotatedTestClass> clazz = DescriptionAnnotatedTestClass.class;
+        final DescriptionInfo descriptionInfo = findDescription(clazz);
+        assertThat(descriptionInfo.docString())
+                .isEqualTo("This is a super description from the annotation");
+    }
+
+    @Description(ANNOTATED_DESCRIPTION_SUPER)
+    private interface DescriptionAnnotatedTestInterface {
+        void testMethod3(String param);
+
+        @Description(ANNOTATED_DESCRIPTION_SUPER)
+        void testMethod4(@Description(ANNOTATED_DESCRIPTION_SUPER) String param);
```

Comment:
Could you add other parameters to make sure if it works with multiple parameters?

Reply:
Thanks. Added one more parameter for each method case.